frontend/src/pages/offers/ViewOfferPage.tsx

```diff
@@ -0,0 +1,61 @@
+import { FunctionComponent } from 'react'
+import { useParams } from 'react-router-dom'
+import ReadOnlyField from '../../components/forms/ReadOnlyField'
+import InternalLink from '../../components/InternalLink'
+import LayoutWithNav from '../../layouts/LayoutWithNav'
+import { useOfferQuery, useShipmentQuery } from '../../types/api-types'
+import { formatShipmentName } from '../../utils/format'
+import { shipmentViewOffersRoute } from '../../utils/routes'
+
+const ViewOfferPage: FunctionComponent = () => {
+  const params = useParams<{ shipmentId: string; offerId: string }>()
+  const offerId = parseInt(params.offerId, 10)
+  const shipmentId = parseInt(params.shipmentId, 10)
+
+  // Figure out the shipment ID from the path
+  const { data: targetShipment } = useShipmentQuery({
+    variables: { id: shipmentId },
+  })
+
+  const { data } = useOfferQuery({ variables: { id: offerId } })
```

Comment:
question (non-blocking): how should we handle 404s?

Reply:
Good point! #102